hyperSpec/R/show.R

```diff
@@ -219,3 +219,117 @@ setMethod("as.character",
+  context("show")
+
+  test_that("show() works", {
+    expect_output(show(faux_cell))
+  })
+
+  test_that("show() gives correct output", {
+    expect_output(show(faux_cell), "hyperSpec object")
+    expect_output(show(faux_cell), "300 data points / spectrum")
+  })
+
+  test_that("show() does not give certain output", {
+    # Does not contain line starting with "wavelength:"
+    expect_output(show(faux_cell), "^(?!wavelength: ).*", perl = TRUE)
+    # Does does not contain line starting with "data:"
+    expect_output(show(faux_cell), "^(?!data: )", perl = TRUE)
+  })
+}
+
+
+.test(print) <- function() {
+  context("print")
+
+  test_that("print() works", {
+    expect_output(print(faux_cell))
+  })
+
+  test_that("print() gives correct output", {
+    expect_output(print(faux_cell), "hyperSpec object")
+    expect_output(print(faux_cell), "300 data points / spectrum")
+
+    expect_output(print(faux_cell, range = TRUE, include = "data"), " rng ")
+    expect_output(print(faux_cell, include = "wl"), "^wavelength: ")
+
+  })
+
+  test_that("print() does not give certain output", {
+    # Does not contain line starting with "wavelength:"
+    expect_output(print(faux_cell), "^(?!wavelength: )", perl = TRUE)
+    # Does does not contain line starting with "data:"
+    expect_output(show(faux_cell), "^(?!data: )", perl = TRUE)
+
+    expect_output(print(faux_cell, include = "data"), "^(?!hyperSpec object)", perl = TRUE)
+    expect_output(print(faux_cell, include = "data"), "^(?!wavelength:)", perl = TRUE)
+
+    expect_output(print(faux_cell, include = "wl"), "^(?!hyperSpec object)", perl = TRUE)
+    expect_output(print(faux_cell, include = "wl"), "^(?!data: )", perl = TRUE)
+  })
+}
+
+
+.test(summary) <- function() {
+  context("summary")
+
+  test_that("summary() works", {
+    expect_output(summary(faux_cell))
+  })
+
+  test_that("summary() gives correct output", {
+    expect_output(summary(faux_cell), "hyperSpec object")
+    expect_output(summary(faux_cell), "300 data points / spectrum")
+    expect_output(summary(faux_cell), "wavelength: ")
+    expect_output(summary(faux_cell), "data: ")
+
+    expect_output(summary(faux_cell, range = TRUE, include = "data"), " rng ")
+  })
+
+# }
+#
+#
+#
+# .test(as.character) <- function() {
+  context("as.character")
+
+  test_that("as.character() works", {
+    expect_is(as.character(faux_cell), "character")
+  })
+
+  test_that("as.character() gives correct output", {
+    set.seed(1)
+    res <- as.character(flu)
+
+    expect_length(res, 9)
+    expect_match(res[1], "hyperSpec object")
+    expect_match(res[4], "181 data points / spectrum")
+    expect_match(res[5], "wavelength: ")
+    expect_match(res[6], "data: ")
+    expect_equal(res[7], "   1. spc: I[fl]/\"a.u.\" [matrix, array181] 27.15000 66.80133 ... 294.6495 ")
+    expect_equal(res[8], "   2. filename: filename [character] rawdata/flu1.txt rawdata/flu2.txt ... rawdata/flu6.txt ")
+    expect_match(res[9], "\\[numeric\\]")
```

Comment:
Nice work @GegznaV !
However, this unit test would fail as soon as we transit to printing with `glimpse` or whatever you suggested in #210 .

Reply:
The point was to create several more general and a few more specific tests. When new changes of code are introduced, the unit tests should be updated accordingly. More specific tests make code more robust to unexpected behavior. 
I think, your (#213) and my (#211) changes should be merged into one PR.

Reply:
I updated this in accordance with what we discussed on Monday (i.e., the dataset used in the tests is created on-the-fly).